src/Slices/ServiceInventory/UI/ServiceInventory.test.tsx

```diff
@@ -144,70 +88,63 @@ function setup(service = Service.a, pageSize = "") {
     scheduler,
   };
 }
+const server = setupServer();
+
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+test("ServiceInventory shows empty view instances", async () => {
+  server.use(
+    http.get("/lsm/v1/service_inventory/service_name_a", () => {
+      return HttpResponse.json({ data: [], metadata: Pagination.metadata });
+    }),
+  );
 
-test("ServiceInventory shows updated instances", async () => {
-  const { component, apiHelper, scheduler } = setup();
+  const { component } = setup();
 
   render(component);
 
   expect(
     await screen.findByRole("region", { name: "ServiceInventory-Loading" }),
   ).toBeInTheDocument();
 
-  apiHelper.resolve(
-    Either.right({
-      data: [],
-      links: Pagination.links,
-      metadata: Pagination.metadata,
-    }),
-  );
-
   expect(
     await screen.findByRole("generic", { name: "ServiceInventory-Empty" }),
   ).toBeInTheDocument();
-
-  scheduler.executeAll();
-
-  apiHelper.resolve(
-    Either.right({
-      data: [ServiceInstance.a],
-      links: Pagination.links,
-      metadata: Pagination.metadata,
-    }),
-  );
-
-  expect(
-    await screen.findByRole("grid", { name: "ServiceInventory-Success" }),
-  ).toBeInTheDocument();
-
-  await act(async () => {
-    const results = await axe(document.body);
-
-    expect(results).toHaveNoViolations();
-  });
 });
 
 test("ServiceInventory shows error with retry", async () => {
-  const { component, apiHelper } = setup();
+  let queryCount = 0;
+
+  server.use(
+    http.get("/lsm/v1/service_inventory/service_name_a", () => {
```

Comment:
i don't mind this implementation, but wonder if there's no other way that is advised by React Query

Reply:
As far as I read the best approach is to mock backend, I think I could here assert for parameter, and Spy on useGet, I'll do that